### PR TITLE
Fix no-tls1_2 and no-dtls1_2

### PR DESCRIPTION
**Pull request by mattcaswell**
##### Checklist

- [x] tests are added or updated

##### Description of change

This fixes no-tls1_2 and no-dtls1_2 configurations. The commit changing sslapitest is for master only. The other commit is for both master and 1.1.0.

### CODE REVIEW

File: test/sslapitest.c [hunk 1]
```diff
@@ -1568,7 +1570,9 @@ int test_main(int argc, char *argv[])
 #ifndef OPENSSL_NO_TLS1_3
     ADD_TEST(test_keylog_no_master_key);
 #endif
+#ifndef OPENSSL_NO_TLS1_2
     ADD_TEST(test_early_cb);
+#endif
```
Comment (levitte) [inline test/sslapitest.c]: So...  this works with SSL v3, TLS v1.0, v1.1 and v1.3, but not v1.2?

Comment (mattcaswell) [inline test/sslapitest.c]: No - this test *only* works if we have v1.2, i.e. if 1.2 is disabled we shouldn't run it. It uses some 1.2 only ciphersuites.

Comment (levitte) [inline test/sslapitest.c]: I read the directive wrong, sorry.  Thanks for explaining.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**